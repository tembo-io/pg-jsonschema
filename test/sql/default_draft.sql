-- Make sure the DSO is loaded.
SELECT jsonschema_is_valid('{"type": "object"}'::json);

-- Should default to 2020
SELECT current_setting('jsonschema.default_draft');
SHOW jsonschema.default_draft;

-- Set to 2019
SELECT set_config('jsonschema.default_draft', 'V2019', false);
SELECT current_setting('jsonschema.default_draft');
SHOW jsonschema.default_draft;

-- Set to v7
SELECT set_config('jsonschema.default_draft', 'V7', false);
SELECT current_setting('jsonschema.default_draft');
SHOW jsonschema.default_draft;

-- Set to v6
SELECT set_config('jsonschema.default_draft', 'V6', false);
SELECT current_setting('jsonschema.default_draft');
SHOW jsonschema.default_draft;

-- Set to v4
SELECT set_config('jsonschema.default_draft', 'V4', false);
SELECT current_setting('jsonschema.default_draft');
SHOW jsonschema.default_draft;

-- Set to invalid value
SELECT set_config('jsonschema.default_draft', 'Nope', false);
SELECT current_setting('jsonschema.default_draft');
SHOW jsonschema.default_draft;
