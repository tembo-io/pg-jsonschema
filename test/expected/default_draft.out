-- Make sure the DSO is loaded.
SELECT jsonschema_is_valid('{"type": "object"}'::json);
 jsonschema_is_valid 
---------------------
 t
(1 row)

-- Should default to 2020
SELECT current_setting('jsonschema.default_draft');
 current_setting 
-----------------
 V2020
(1 row)

SHOW jsonschema.default_draft;
 jsonschema.default_draft 
--------------------------
 V2020
(1 row)

-- Set to 2019
SELECT set_config('jsonschema.default_draft', 'V2019', false);
 set_config 
------------
 V2019
(1 row)

SELECT current_setting('jsonschema.default_draft');
 current_setting 
-----------------
 V2019
(1 row)

SHOW jsonschema.default_draft;
 jsonschema.default_draft 
--------------------------
 V2019
(1 row)

-- Set to v7
SELECT set_config('jsonschema.default_draft', 'V7', false);
 set_config 
------------
 V7
(1 row)

SELECT current_setting('jsonschema.default_draft');
 current_setting 
-----------------
 V7
(1 row)

SHOW jsonschema.default_draft;
 jsonschema.default_draft 
--------------------------
 V7
(1 row)

-- Set to v6
SELECT set_config('jsonschema.default_draft', 'V6', false);
 set_config 
------------
 V6
(1 row)

SELECT current_setting('jsonschema.default_draft');
 current_setting 
-----------------
 V6
(1 row)

SHOW jsonschema.default_draft;
 jsonschema.default_draft 
--------------------------
 V6
(1 row)

-- Set to v4
SELECT set_config('jsonschema.default_draft', 'V4', false);
 set_config 
------------
 V4
(1 row)

SELECT current_setting('jsonschema.default_draft');
 current_setting 
-----------------
 V4
(1 row)

SHOW jsonschema.default_draft;
 jsonschema.default_draft 
--------------------------
 V4
(1 row)

-- Set to invalid value
SELECT set_config('jsonschema.default_draft', 'Nope', false);
ERROR:  invalid value for parameter "jsonschema.default_draft": "Nope"
HINT:  Available values: V4, V6, V7, V2019, V2020.
SELECT current_setting('jsonschema.default_draft');
 current_setting 
-----------------
 V4
(1 row)

SHOW jsonschema.default_draft;
 jsonschema.default_draft 
--------------------------
 V4
(1 row)

